### ISSUE TITLE
Nullable value type warnings don't follow flow analysis for re-assignment?

### ISSUE THREAD

**Issue by Pilchie** [dotnet/roslyn#31871]
**Version Used**: 
3.0.0-beta2-18616-01

**Steps to Reproduce**:
```C#
using System;

class C
{
    bool IsEnabled(string variable, ref bool? result)
    {
        if (result == null)
        {
            var value = "";
            result = string.Equals(value, "1", StringComparison.OrdinalIgnoreCase);
        }

        return result.Value;
    }
}
```

**Expected Behavior**:
No warning

**Actual Behavior**:
Program.cs(13,16,13,28): warning CS8629: Nullable value type may be null.

**Note**:
May be fixed in latest compiler.  I didn't see this in the dogfood VS, but I did with the package above.  Or the package above is newer than the VS?

**Comment by Pilchie** [comment]
Tag @cston

**Comment by cston** [comment]
Note: The warning is not reported if `result` is passed by value.

**Comment by jaredpar** [comment]
Possible dupe of #29954 (CC @jcouv)

**Comment by jcouv** [comment]
Correct, it's a dupe. I'll go ahead and close. Thanks
